core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -149,12 +149,25 @@ class LogCleaner(val config: CleanerConfig,
   }
 
   /**
-   * For testing, a way to know when work has completed. This method blocks until the
+   * For testing, a way to know when work has completed. This method waits until the
    * cleaner has processed up to the given offset on the specified topic/partition
+   *
+   * @param topic The Topic to be cleaned
+   * @param part The partition of the topic to be cleaned
+   * @param offset The first dirty offset that the cleaner doesn't have to clean
+   * @param maxWaitMs The maximum time in ms to wait for cleaner
+   *
+   * @return A boolean indicating whether the work has completed before timeout
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
-      Thread.sleep(10)
+  def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
```

Comment:
`sleepInterval = 100` is better if you want to make it possible for callers to configure that parameter. It seems to me that it's unnecessary, but up to you. It is not the worst thing in the world to have to change two things in adjacent lines, so you can also keep it as is if committers are fine with the code (there is also a slight readability penalty as users have to figure out that the `100` in both cases represent the same thing). It _is_ a minor thing for sure, but personally I try to write code so that there are no minor things when the solution is easy. :)

I think I didn't explain myself well regarding point 2, but let's just drop it as it's subtle and it's not worth spending the time discussing it in this context.